### ISSUE TITLE
a11y announcements of fancy text field

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#15187]
Internal: b/292548468

![image](https://user-images.githubusercontent.com/1227763/37008104-777c765a-2095-11e8-938e-88fe4865ec3a.png)

What is the a11y behavior of the fancy text fields supposed to be that have a placeholder text, a helper, and a character count? What should be announced how?

Also, what about the text field that has a suffix with "/dozen"??

How should errors be announced?

Needs to be defined for iOS and Android.

@HansMuller

**Comment by goderbauer** [comment]
Looks like this can get even more complicated: 
![image](https://user-images.githubusercontent.com/1227763/37008623-e021c672-2097-11e8-87eb-9ded889577a2.png)

**Comment by jonahwilliams** [comment]
@goderbauer what do you think about just writing a custom render object to handle semantics?

**Comment by goderbauer** [comment]
Can you elaborate on what that custom render object would do with regards to semantics?

**Comment by jonahwilliams** [comment]
I've made some progress on this, but will likely need to fallback to using reflection to get the hint order correct on Android.
